### PR TITLE
Use GLib idle_add instead of Mainloop in async task

### PR DESCRIPTION
**Pull request by rmacqueen**
Mainloop.idle_add always adds its callbacks with DEFAULT
priority. In the capacitate app, this has resulted
in cases where the callback is not invoked if many of
them are queued at once. Instead we should use GLib.idle_add,
which allows us to specify a higher priority.

https://phabricator.endlessm.com/T14072

### CODE REVIEW

**Comment by ptomato** [comment]
Looks like it will work fine in your case, but what effects will it have on other code using Async.Task?

I would prefer to add an argument to `_init()` such as `priority=GLib.PRIORITY_DEFAULT` so that you can determine the priority when you construct the task.

**Comment by ptomato** [comment]
You can remove the `Mainloop` import now.

**Comment by rmacqueen** [comment]
@ptomato come to think of it, why are we even having to call this function on the idle thread? Why can't we just call it directly?

File: js/search/asyncTask.js [hunk 1]
```diff
@@ -26,6 +25,20 @@ const AsyncTask = Lang.Class({
     GTypeName: 'EknAsyncTask',
     Extends: GObject.Object,
 
+    Properties: {
```
Comment (ptomato) [inline js/search/asyncTask.js]: Since the other stuff (`source, cancellable, callback`) is all passed directly to the constructor instead of through a properties dict, I think it would be best to do the same with priority. You could provide a default value for the argument. In any case it won't work as-is, because the `this.parent()` call has to get a properties dict for this property to end up set.

That way you wouldn't have to specify a range for the priority either, which is more in line with what GLib does — you can set any integer as the priority.

I would also make the default value `GLib.PRIORITY_DEFAULT`, I don't think it would be good for all tasks to always run on high priority.

File: js/search/xapianBridge.js [hunk 2]
```diff
@@ -118,7 +118,7 @@ const XapianBridge = new Lang.Class({
     // Queues a SoupMessage for *uri* to the current http session. Calls
     // *callback* on any errors encountered and the parsed JSON.
     _send_json_ld_request: function (uri, cancellable, callback) {
-        let task = new AsyncTask.AsyncTask(this, cancellable, callback);
+        let task = new AsyncTask.AsyncTask(this, cancellable, callback, GLib.PRIORITY_HIGH);
```
Comment (ptomato) [inline js/search/xapianBridge.js]: Could you please split this into a separate commit than the feature itself?
